### ISSUE TITLE
vsphere_virtual_machine - Bios disk boot order changes when adding disk to SCSI controller 3

### ISSUE THREAD

**Issue by tigerhawk6** [terraform-providers/terraform-provider-vsphere#637]
### Terraform Version

Terraform v0.11.8

### vSphere Provider Version

provider.vsphere v1.8.1

### Affected Resource(s)

- `vsphere_virtual_machine`

### Terraform Configuration Files

```hcl
provider "vsphere" {
    version         = "~> 1.8"
    user            = "${var.vsphere_user}"
    password        = "${var.vsphere_password}"
    vsphere_server  = "${var.vsphere_server}"
    
    allow_unverified_ssl = true
}

data "vsphere_datacenter" "datacenter" {
  name = "${var.vsphere_dc}"
}

data "vsphere_datastore" "datastore" {
  name          = "${var.vsphere_datastore}"
  datacenter_id = "${data.vsphere_datacenter.datacenter.id}"
}

data "vsphere_network" "network1" {
  name          = "${var.client_network"
  datacenter_id = "${data.vsphere_datacenter.datacenter.id}"
}

data "vsphere_virtual_machine" "template" {
  name          = "${var.client_template}"
  datacenter_id = "${data.vsphere_datacenter.datacenter.id}"
}

resource "vsphere_folder" "folder" {
  path          = "${var.client_name}"
  type          = "vm"
  datacenter_id = "${data.vsphere_datacenter.datacenter.id}"
}
data "vsphere_compute_cluster" "cluster" {
  name          = "${var.client_cluster"
  datacenter_id = "${data.vsphere_datacenter.datacenter.id}"
}

resource "vsphere_virtual_machine" "testDB" {
  count                 = "${var.rac_node_number}"
  name                  = "${var.client_name}-db${count.index + 1}"
  resource_pool_id      = "${data.vsphere_compute_cluster.cluster.resource_pool_id}"
  datastore_id          = "${data.vsphere_datastore.datastore.id}"
  folder                = "${vsphere_folder.folder.path}"
  
  num_cpus              = "${var.rac_vcpu}"
  num_cores_per_socket  = "${var.rac_vcpu / 2}"
  memory                = "${var.rac_mem}"
  guest_id              = "${data.vsphere_virtual_machine.template.guest_id}"
  scsi_type             = "${data.vsphere_virtual_machine.template.scsi_type}"
  scsi_controller_count = "4"
  
  network_interface {
    network_id          = "${data.vsphere_network.network1.id}"
    adapter_type        = "${data.vsphere_virtual_machine.template.network_interface_types[0]}"
  }

  disk {
    label             = "disk0"
    size              = 20
    thin_provisioned  = false
    unit_number       = 0
  }
  
  disk {
    label             = "disk1"
    size              = 20
    thin_provisioned  = false
    unit_number       = 35
  }

  disk {
    label             = "disk2"
    size              = 20
    thin_provisioned  = false
    unit_number       = 50
  }

  clone {
    template_uuid = "${data.vsphere_virtual_machine.template.id}"

    customize {
      linux_options {
        host_name = "${var.client_name}-db${count.index + 1}"
        domain    = "${var.client_name}.${var.root_zone}"
        time_zone = "America/Chicago"
      }
      network_interface {}
    }
  }
}
```

### Debug Output

[terraform.log](https://github.com/terraform-providers/terraform-provider-vsphere/files/2457244/terraform.log)

### Expected Behavior

* Clone VM based on submitted template variable
* Boot disk is on SCSI controller 0
* Add 3 more SCSI controllers
* Created 2 new disks
 * Add first one to SCSI Controller 2
 * Add second one to SCSI Controller 3

This is a working build process. Will eventually have ~24 disks spread across all controllers for performance. 

### Actual Behavior

* VM is cloned and disk created
* VM Bios changed so that disk on SCSI 3:5 is ahead of 0:0 in boot order
* VM tries to PXE boot since it can't boot from boot drive 0:0
* TF times out since VM doesn't complete build/customization process.

Screenshot of TF created VM on the left and manually created VM on the right.
![bios_boot](https://user-images.githubusercontent.com/24719768/46627566-79e09500-cb00-11e8-90d1-dd30c1bcbd82.PNG)

### Steps to Reproduce

Please list the steps required to reproduce the issue, for example:
1. `terraform apply`
2. Open console on VM once deployed
3. Reboot VM, hit escape QUICKLY to get boot options
4. Select SETUP
5. Go to Boot tab
6. Go to Hard Drive and hit Enter

### Important Factoids

* Tested against vSphere 6.5, 6.7 and 6.8 (VMC)
* Tested with RHEL, CentOS, and Windows 2012R2 VM templates.

### References

Couldn't find any.

**Comment by tigerhawk6** [comment]
Changed to an Oracle Enterprise Linux template and had the same issue but this time with the 3rd SCSI controller (SCSI IDs 2:0-2:15). I can use the 4th controller just fine, though. This is WEIRD.

**Comment by tigerhawk6** [comment]
I did some more testing and believe this is an OS issue.

If I add all 4 SCSI controllers BEFORE installing its OS then I have no problems attaching drives to ANY controller.

If I add the SCSI controllers at any point AFTER installing the OS, I always get hard drives set as a higher boot order than 0:0 within the BIOS from at least one controller.

I've validated this against multiple VMHW versions, vCenter, ESXi, OS versions.  Can someone else validate this as well? If so I'm good with closing this.

**Comment by tigerhawk6** [comment]
Bill-

I did some more testing this weekend, verified again this morning. Here's what I've found:

1. If I create the "base" VM via vSphere UI or PowerCLI and DO NOT add the addition SCSI controllers before loading the OS, I have issues with the bios boot order upon TF deployment.
2. If I create the "base" VM via vSphere UI or PowerCLI and DO NOT add the addition SCSI controllers before loading the OS, I can use vSphere UI or PowerCLI to deploy new VMs and use "customize hardware" to added controllers & disks. This works successfully.
3. If I create the "base" VM via vSphere UI or PowerCLI and ADD the SCSI controllers before loading the OS, then TF deployments work properly. 
4. If I create the "base" VM via Packer with NO addition SCSI controllers (can't be added via Packer), then TF deployments work properly, including adding the controllers via TF.

I've compared VMX files across all 3. Only difference I see is that packer adds fewer items, but nothing that would lead me to think it would affect Bios boot order.

We're using Packer to create base images in AWS/GCP so using this method will work for us.  I just think it is VERY weird
